sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala

```diff
@@ -258,16 +258,20 @@ class RapidsShuffleIterator(
                   blockManagerId, id, mapIndex, errorMessage, throwable))
               }
             }
+
+            // tell the client to cancel pending requests
+            client.cancelPending(this)
+          }
         }
 
         logInfo(s"Client $blockManagerId triggered, for ${shuffleRequestsMapIndex.size} blocks")
         client.doFetch(shuffleRequestsMapIndex.map(_.id), handler)
-        clients = clients :+ client
+        clientAndHandlers = clientAndHandlers :+ ((client, handler))
```

Comment:
```suggestion
        clientAndHandlers = clientAndHandlers :+ (client, handler)
```

Reply:
In this case, `clientAndHandlers` is a `Seq` of tuple, so the original signature is required according to my compiler.

Reply:
Weird, it works for me in the Spark Scala REPL:
```
scala> var x: Seq[(Int, Int)] = Nil
x: Seq[(Int, Int)] = List()

scala> x = x :+ (1, 2)
x: Seq[(Int, Int)] = List((1,2))

scala> x = x :+ ((1, 2))
x: Seq[(Int, Int)] = List((1,2), (1,2))

scala> x = x :+ (1, 2)
x: Seq[(Int, Int)] = List((1,2), (1,2), (1,2))
```

The `:+` operator adds a single object to the sequence, and what's being added is a tuple.  Oh well no worries, if it's necessary it's necessary.

Reply:
I can reproduce your case in the scala repl. But when I try with the `Seq` in the iterator, this is what I get:

```
diff --git a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala
index 969f2c955..8178ef664 100644
--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala
@@ -266,7 +266,7 @@ class RapidsShuffleIterator(
 
         logInfo(s"Client $blockManagerId triggered, for ${shuffleRequestsMapIndex.size} blocks")
         client.doFetch(shuffleRequestsMapIndex.map(_.id), handler)
-        clientAndHandlers = clientAndHandlers :+ ((client, handler))
+        clientAndHandlers = clientAndHandlers :+ (client, handler)
       }
     }
```

```
[ERROR] [Error] /home/abellina/work/rapids-plugin-4-spark-latest/sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala:269: too many arguments (2) for method :+: (elem: (com.nvidia.spark.rapids.shuffle.RapidsShuffleClient, com.nvidia.spark.rapids.shuffle.RapidsShuffleFetchHandler))(implicit bf: scala.collection.generic.CanBuildFrom[Seq[(com.nvidia.spark.rapids.shuffle.RapidsShuffleClient, com.nvidia.spark.rapids.shuffle.RapidsShuffleFetchHandler)],(com.nvidia.spark.rapids.shuffle.RapidsShuffleClient, com.nvidia.spark.rapids.shuffle.RapidsShuffleFetchHandler),That])That
```